conan/tools/env/environment.py

```diff
@@ -0,0 +1,291 @@
+        :type other: Environment
+        """
+        for k, v in other._values.items():
+            existing = self._values.get(k)
+            if existing is None:
+                self._values[k] = v
+            else:
+                try:
+                    index = v.index(_EnvVarPlaceHolder)
+                except ValueError:  # The other doesn't have placeholder, overwrites
+                    self._values[k] = v
+                else:
+                    new_value = v[:]  # do a copy
+                    new_value[index:index + 1] = existing  # replace the placeholder
+                    # Trim front and back separators
+                    val = new_value[0]
+                    if isinstance(val, _Sep) or val is _PathSep:
+                        new_value = new_value[1:]
+                    val = new_value[-1]
+                    if isinstance(val, _Sep) or val is _PathSep:
+                        new_value = new_value[:-1]
+                    self._values[k] = new_value
+        return self
+
+
+class ProfileEnvironment:
+    def __init__(self):
+        self._environments = OrderedDict()
+
+    def __repr__(self):
+        return repr(self._environments)
+
+    def get_env(self, ref):
+        """ computes package-specific Environment
+        it is only called when conanfile.buildenv is called
+        """
+        result = Environment()
+        for pattern, env in self._environments.items():
+            if pattern is None or fnmatch.fnmatch(str(ref), pattern):
+                env = self._environments[pattern]
+                result = result.compose(env)
+        return result
+
+    def compose(self, other):
+        """
+        :type other: ProfileEnvironment
+        """
+        for pattern, environment in other._environments.items():
+            existing = self._environments.get(pattern)
+            if existing is not None:
+                self._environments[pattern] = existing.compose(environment)
+            else:
+                self._environments[pattern] = environment
+
+    @staticmethod
+    def loads(text):
+        result = ProfileEnvironment()
+        for line in text.splitlines():
+            line = line.strip()
+            if not line or line.startswith("#"):
+                continue
+            for op, method in (("+=", "append"), ("=+", "prepend"),
+                               ("=!", "unset"), ("=", "define")):
+                tokens = line.split(op, 1)
+                if len(tokens) != 2:
+                    continue
+                pattern_name, value = tokens
+                pattern_name = pattern_name.split(":", 1)
+                if len(pattern_name) == 2:
+                    pattern, name = pattern_name
+                else:
+                    pattern, name = None, pattern_name[0]
+
+                env = Environment()
+                if method == "unset":
+                    env.unset(name)
+                else:
+                    if value.startswith("(path)"):
```

Comment:
I tried a few alternatives, like different operators, but this was the most readable and intuitive and easy to parse at the same time, I think it is difficult to beat this, but suggestions welcome:
```
# define
MyPath1=(path)/my/path1
#  append
MyPath2+=(path)/my/path2
```

Reply:
My concern is that `(path)` is valid for a path name... and if something can happen, it will happen 😄  

```
⇒  pwd
/Users/jgsogo/dev/conan/tmp/path(weird/(path)porqueyolovalgo
[py38]jgsogo@Javierg:~/dev/conan/tmp/path(weird/(path)porqueyolovalgo|
```

maybe we need to rely on comments?

```
# define
MyPath1=/my/path1  # path, type:path
#  append
MyPath2+=/my/path2   # path, type:path,...
```

Reply:
The problem is that there might be env-vars that need to contain a # too, so lines cannot contain comments either.

Strictly speaking, the only character that cannot be an env-var name is ``=``. Even the ``+=`` or ``=!`` could be colliding with values or variable names containing those special characters.
But in practice, POSIX shells only allow alphanumeric identifiers, so it is relatively extended practice. 

As the environment variable values can contain any arbitrary value, then there is by definition impossible to have any ``VAR OPERATION VALUE`` definition that implements the different logic operations and completely avoid all possible collisions. We would need a structured format, json, xml, or equivalent, but I doubt we want to go that way.

If we are concerned about ``(path)`` colliding, the best action could be to make the token more specific, something like ``var+=(conanpath!)/path/to/something``